lib/chezmoi/boltpersistentstate.go

```diff
@@ -87,9 +87,10 @@ func (b *BoltPersistentState) Get(bucket, key []byte) ([]byte, error) {
 // it does not already exist.
 func (b *BoltPersistentState) Set(bucket, key, value []byte) error {
 	if b.db == nil {
-		if err := b.openDB(); err != nil {
+		if err := b.openDB(true); err != nil {
 			return err
 		}
+		defer b.Close()
```

Comment:
I don't think this is correct: it causes the database to be opened and closed on every `Set`.

Reply:
This just mimics the previous behavior. If the database gets opened automatically, it should also get closed automatically imo. If it's open already, that defer statement won't get called.

Reply:
The original intent is that the database should be opened lazily, i.e. only when accessed, but once opened it should stay open. I'll revisit this logic later.